aiida/engine/processes/ports.py

```diff
@@ -99,6 +101,21 @@ class InputPort(WithSerialize, WithNonDb, ports.InputPort):
     value serialization to database storable types and support non database storable input types as well.
     """
 
+    def __init__(self, name, valid_type=None, help=None, default=ports.UNSPECIFIED, required=True, validator=None):
+        """Override the constructor to check the type of the default if set and warn if not immutable."""
+        # pylint: disable=redefined-builtin,too-many-arguments
+        immutabe_types = (int, float, complex, tuple, frozenset, bytes, six.string_types)
```

Comment:
... and: typo, should be `immutable_types`

Reply:
But I agree with @ltalirz: needs documentation, and I maybe implement it as a static function from the start since a candidate from Python 3.7 would be `@dataclasses.dataclass(frozen=True)` decorated class?
There you don't get a type you could check against but would have to do a separate check (`if isdataclass(...) and hasattr(..., '__setattr__')`).

Reply:
I couldn't find any definitive resource in the official python docs. I agree that this is fragile.